src/pylero/test_run.py

```diff
@@ -722,9 +723,12 @@ def _status_change(self):
         else:
             status = "inprogress"
             check_tr.finished_on = None
-        if status != check_tr.status:
+        # Before setting the status, check if the new value is valid
+        status_id = self._cls_suds_map["status"]["enum_id"]
+        valid_status_values = self.get_valid_field_values(status_id, None)
+        if status != check_tr.status and status in valid_status_values:
             check_tr.status = status
-            check_tr.update()
+        check_tr.update()
```

Comment:
- yes, it is the original function of `_change_status`.
- IMHO, I think it would be cleaner to set `finished_on` in a new method. Then `_change_status` is doing what and only what the name implies.
- What are your thoughts- should I refactor this?

PS: Sorry for the late reply on my end.

Reply:
The original function only do the update if the status has changed, else it will not update the finished_on value.

The finished_on and status is correlated in logic (reset to None when status is notrun or inprogress), so better to be updated together.

I don't think it need to be separated into a new function for the finished_on value.

Since there is no specific reason to update the finshed_on value, we don't need update line 731.

Reply:
Thanks for clarifying this for me. I will move the update into the if condition :)